pkg/nsx/services/ippool/ippool.go

```diff
@@ -0,0 +1,262 @@
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{common.TagScopeIPPoolCRUID: indexFunc}),
+		BindingType: model.IpAddressPoolBindingType(),
+	}}
+	ipPoolService.ipPoolBlockSubnetStore = &IPPoolBlockSubnetStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{common.TagScopeIPPoolCRUID: indexFunc}),
+		BindingType: model.IpAddressPoolBlockSubnetBindingType(),
+	}}
+
+	go ipPoolService.InitializeResourceStore(&wg, fatalErrors, ResourceTypeIPPool, nil, ipPoolService.ipPoolStore)
+	tags := []model.Tag{
+		{Scope: String(common.TagScopeIPSubnetOwner), Tag: String(ResourceTypeIPPool)},
+	}
+	go ipPoolService.InitializeResourceStore(&wg, fatalErrors, ResourceTypeIPPoolBlockSubnet, tags, ipPoolService.ipPoolBlockSubnetStore)
+
+	go func() {
+		wg.Wait()
+		close(wgDone)
+	}()
+	select {
+	case <-wgDone:
+		break
+	case err := <-fatalErrors:
+		close(fatalErrors)
+		return ipPoolService, err
+	}
+	return ipPoolService, nil
+}
+
+func (service *IPPoolService) CreateOrUpdateIPPool(obj *v1alpha2.IPPool) (bool, bool, error) {
+	nsxIPPool, nsxIPSubnets := service.BuildIPPool(obj)
+	existingIPPool, existingIPSubnets, err := service.indexedIPPoolAndIPPoolSubnets(obj.UID)
+	if err != nil {
+		log.Error(err, "failed to get ip pool and ip pool subnets by UID", "UID", obj.UID)
+		return false, false, err
+	}
+	log.V(1).Info("existing ippool and ip subnets", "existingIPPool", existingIPPool, "existingIPSubnets", existingIPSubnets)
+	ipPoolSubnetsUpdated := false
+	ipPoolUpdated := common.CompareResource(IpAddressPoolToComparable(existingIPPool), IpAddressPoolToComparable(nsxIPPool))
+	changed, stale := common.CompareResources(IpAddressPoolBlockSubnetsToComparable(existingIPSubnets), IpAddressPoolBlockSubnetsToComparable(nsxIPSubnets))
+	changedIPSubnets, staleIPSubnets := ComparableToIpAddressPoolBlockSubnets(changed), ComparableToIpAddressPoolBlockSubnets(stale)
+	for i := len(staleIPSubnets) - 1; i >= 0; i-- {
+		staleIPSubnets[i].MarkedForDelete = &MarkedForDelete
+	}
+	finalIPSubnets := append(changedIPSubnets, staleIPSubnets...)
+	if len(finalIPSubnets) > 0 {
+		ipPoolSubnetsUpdated = true
+	}
+
+	if err := service.Operate(nsxIPPool, finalIPSubnets, ipPoolUpdated, ipPoolSubnetsUpdated); err != nil {
+		return false, false, err
+	}
+
+	realizedSubnets, subnetCidrUpdated, e := service.AcquireRealizedSubnetIP(obj)
+	if e != nil {
+		return false, false, e
+	}
+	obj.Status.Subnets = realizedSubnets
+	return subnetCidrUpdated, ipPoolSubnetsUpdated, nil
+}
+
+func (service *IPPoolService) Operate(nsxIPPool *model.IpAddressPool, nsxIPSubnets []*model.IpAddressPoolBlockSubnet, IPPoolUpdated bool, IPPoolSubnetsUpdated bool) error {
+	if !(IPPoolUpdated || IPPoolSubnetsUpdated) {
+		return nil
+	}
+	infraIPPool, err := service.WrapHierarchyIPPool(nsxIPPool, nsxIPSubnets)
+	if err != nil {
+		return err
+	}
+	// Get IPPool Type from nsxIPPool
+	IPPoolType := common.IPPoolTypePublic
```

Comment:
in IPPool CR, the default value for type is Private, so change #113 to Private.

Reply:
ok